var/spack/repos/builtin/packages/mfem-cmake/package.py

```diff
@@ -508,6 +515,11 @@ def find_optional_library(name, prefix):
             cfg.write(cmake_cache_entry("CMAKE_CUDA_COMPILER",
                                         cudacompiler))
 
+            # JBE: CudaPackage will set this in the environment which overrides
+            # anything else - is this related to https://github.com/spack/spack/issues/17823 ??
+            if '+mpi' in spec:
+                env['CUDAHOSTCXX'] = spec['mpi'].mpicxx
+
```

Comment:
What is happening here?
Does this need to be done because otherwise the host compiler will be set not to the mpi compiler?

Reply:
Yes, I think it might be inaccurate to say that CudaPackage was responsible but I noticed that the CUDAHOSTCXX in the environment (`spack-build-env.txt`) was set to the "base" CXX compiler and not the MPI CXX compiler.  This change appeared to fix the issue.